mycodo/mycodo_flask/routes_method.py

```diff
@@ -58,118 +56,7 @@ def method_data(method_id):
-                [get_sec(each_method.time_start) * 1000,
-                 each_method.setpoint_start])
-            method_list.append(
-                [get_sec(each_method.time_end) * 1000,
-                 setpoint_end])
-            method_list.append(
-                [get_sec(each_method.time_start) * 1000,
-                 None])
-
-    elif method.method_type == "DailyBezier":
-        points_x = 700
-        seconds_in_day = 60 * 60 * 24
-        P0 = (last_method_data.x0, last_method_data.y0)
-        P1 = (last_method_data.x1, last_method_data.y1)
-        P2 = (last_method_data.x2, last_method_data.y2)
-        P3 = (last_method_data.x3, last_method_data.y3)
-        for n in range(points_x):
-            percent = n / float(points_x)
-            second_of_day = percent * seconds_in_day
-            if second_of_day == 0:
-                continue
-            y = bezier_curve_y_out(last_method_data.shift_angle,
-                                   P0, P1, P2, P3,
-                                   second_of_day)
-            method_list.append([percent * seconds_in_day * 1000, y])
-
-    elif method.method_type == "DailySine":
-        points_x = 700
-        seconds_in_day = 60 * 60 * 24
-        for n in range(points_x):
-            percent = n / float(points_x)
-            angle = n / float(points_x) * 360
-            y = sine_wave_y_out(last_method_data.amplitude,
-                                last_method_data.frequency,
-                                last_method_data.shift_angle,
-                                last_method_data.shift_y,
-                                angle)
-            method_list.append([percent * seconds_in_day * 1000, y])
-
-    elif method.method_type == "Duration":
-        first_entry = True
-        start_duration = 0
-        method_data = method_data.filter(MethodData.output_id == None)
-        for each_method in method_data:
-            if each_method.setpoint_end is None:
-                setpoint_end = each_method.setpoint_start
-            else:
-                setpoint_end = each_method.setpoint_end
-
-            if each_method.duration_sec == 0:
-                pass  # Method line is repeat command, don't add to method_list
-            elif first_entry:
-                method_list.append([0, each_method.setpoint_start])
-                method_list.append([each_method.duration_sec, setpoint_end])
-                start_duration += each_method.duration_sec
-                first_entry = False
-            else:
-                end_duration = start_duration + each_method.duration_sec
-
-                method_list.append(
-                    [start_duration, each_method.setpoint_start])
-                method_list.append(
-                    [end_duration, setpoint_end])
-
-                start_duration += each_method.duration_sec
-
-    return jsonify(method_list)
+    return jsonify(method_by_type(method_data, method.method_type).get_plot(700))
```

Comment:
This code has been split up and moved into the different method type implementation classes.